src/portability/portability.md

```diff
@@ -1,3 +1,56 @@
 # Portability
 
-> ❌: This section has not yet been written. Please refer to [rust-embedded/book#6](https://github.com/rust-embedded/book/issues/6) for discussion of this section.
+In embedded environments portability is a very important topic: Every vendor and even each family from a single manufacturer offers different peripherals and capabilities and similarly the ways to interact with the peripherals will vary.
+
+A common way to equalize such differences is via a layer called Hardware Abstraction layer or **HAL**.
+
+> Hardware abstractions are sets of routines in software that emulate some platform-specific details, giving programs direct access to the hardware resources.
+> 
+> They often allow programmers to write device-independent, high performance applications by providing standard operating system (OS) calls to hardware. 
+>
+> *Wikipedia: [Hardware Abstraction Layer]*
+
+[Hardware Abstraction Layer]: https://en.wikipedia.org/wiki/Hardware_abstraction
+
+Embedded systems are a bit special in this regard since we typically do not have operating systems and user installable software but firmware images which are compiled as a whole as well as a number of other constraints. So while the traditional approach as defined by Wikipedia could potentially work it is likely not the most productive approach to ensure portability.
+
+How do we do this in Rust? Enter **embedded-hal**...
+
+## What is embedded-hal?
+
+In a nutshell it is a set of traits which define implementation contracts between **HAL implementations**, **drivers** and **applications (or firmwares)**. Those contracts include both capabilities (i.e. if a trait is implemented for a certain type, it is easy to see that a **HAL implementation** provides a certain capability) and methods (i.e. if you can construct a type implementing a trait it is guarenteed that you have the methods specified in the trait available).
```

Comment:
Suggestion, avoid phrases like "it is easy to see", as it might not be easy for all people